src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -111,16 +119,18 @@ private void EnsureInitialized(WebHostSettings settings)
                     _standbySecretManager = null;
                     _standbyHostManager = null;
                     _standbyReceiverManager = null;
+
+                    _settingsManager.Clear();
```

Comment:
I haven't seen the impl. yet, but SettingsManager.Clear seems strange. I assume you're resetting cached values. Perhaps Reset is a better name? Clear implies some sort of deletion.